### PR TITLE
Pass in options to hook worker (continued)

### LINKED ISSUE #272: Any way of being able to set the HOOK_TIMEOUT value?
Is there any chance that the HOOK_TIMEOUT constant in hooks-worker-client could be part of a config, as some tests may take more than 5 seconds and therefore fail. I'd like a param to be optional that the HOOK_TIMEOUT could be set.

### LINKED ISSUE #328: Make wait time for client TCP server configurable
...because some times it is not able to start within 1 second.

### PR DESCRIPTION
**Pull request by honzajavorek**
This takes @xvalentino's amazing work from https://github.com/apiaryio/dredd/pull/382 and addresses @netmilk's comments.

Closes #328 :tada:
Closes #272 :tada:
Closes #382 :tada:

### CODE REVIEW

**Comment by honzajavorek** [comment]
@netmilk Since you wrote earlier

> I think it's good to merge if we update the documentation on CLI and JS interface and remove the possibility to configure the message delimiter.

I consider this my PR to be good to merge once tests are green. If you find any problems though, feel free to comment and I'll fix them in subsequent PRs.

File: docs/hooks-new-language.md [hunk 1]
```diff
@@ -69,6 +69,21 @@ If you want to write a hook handler for your language you will have to implement
...
+There are several configuration options, which can help you during development:
+
+- `--hooks-worker-timeout`: `5000` - How long to wait for hooks worker to start.
+- `--hooks-worker-connect-timeout`: `1500` - How long to wait for hooks worker to acknowledge connection.
+- `--hooks-worker-connect-retry`: `500` - How long to wait between attempts to connect to hooks worker.
+- `--hooks-worker-after-connect-wait`: `100` - How long to wait between connecting to hooks worker and start of testing.
+- `--hooks-worker-term-timeout`: `5000` - How long to wait between trying to terminate hooks worker and killing it.
+- `--hooks-worker-term-retry`: `500` - How long to wait between attempts to terminate hooks worker.
+- `--hooks-worker-handler-host`: `localhost` - Host of the hook worker.
+- `--hooks-worker-handler-port`: `61321` - Port of the hook worker.
```
Comment (romansklenar) [inline docs/hooks-new-language.md]: Are these supposed to work also from `dredd.yml` config file?

Comment (honzajavorek) [inline docs/hooks-new-language.md]: They should. Not sure if `dredd init` automatically picks them up, but they should definitely work if inserted to the YAML manually. If it doesn't work for you, please file an issue, that would be a bug.

Comment (romansklenar) [inline docs/hooks-new-language.md]: Tried it now - works! :+1:

Comment (honzajavorek) [inline docs/hooks-new-language.md]: :sweat_smile:

**Pull request merged**